lib/BridgeManager.php

```diff
@@ -151,22 +156,23 @@ private function editBridgeConfig(string $token, array $newBridge) {
 	}
 
 	private function addLocalPart($token, $bridge) {
-		$user = $this->checkBotUser($token);
+		$botInfo = $this->checkBotUser($token, $bridge['enabled']);
 		$localPart = [
 			'type' => 'nctalk',
-			'login' => $user['id'],
-			'password' => $user['password'],
+			'login' => $botInfo['id'],
+			'password' => $botInfo['password'],
 			'channel' => $token,
 		];
 		array_push($bridge['parts'], $localPart);
 		return $bridge;
 	}
 
-	private function checkBotUser(string $token) {
+	private function checkBotUser(string $token, bool $create): array {
 		$botUserId = 'matterbridge-bot';
-		// TODO check user exists and create it if necessary
+		// check if user exists and create it if necessary
 		if (!$this->userManager->userExists($botUserId)) {
 			$pass = md5(strval(rand()));
+			$this->config->setAppValue('spreed', 'bot_pass', $pass);
```

Comment:
Make sure to add this to https://github.com/nextcloud/server/blob/cb057829f72c70e819f456edfadbb29d72dba832/lib/private/AppConfig.php#L51-L57

Reply:
Thanks a lot for the review!

I don't know if it even makes sense to store the bot password as I've seen app token generation works whatever 'password' parameter we give it.

Maybe I didn't grasp the meaning of the 'password' parameter when generating an app token. It appears not to be the user password. Do you know what it is used for?

Reply:
I think its needed to be able to unencrypt the user keys in the session. So in case of server side encryption you wouldn't be able to read files. So make sure it's stored and save